### DISCUSSION TITLE
Network Transport security

### DISCUSSION THREAD

**Discussion by wbodley** [eclipse-ecal/ecal#982] [Q&A] [answered]
Hi, 

I'm just wondering if there were any future plans or possible solutions currently that could support some level of TLS transport security over the network, discovery is not really an issue, its really about RPC or pub/sub security.

Thanks,

Will

**Comment by KerstinKeller** [comment, marked as answer]
HI Will,
every now and then we have slightly touched upon this topic, but then quickly agreed that this is out of scope for eCAL at this moment.
eCAL is designed to be used in trusted networks and focusses on the high throughput and low latency.

My knowledge regarding security is very limited. I guess we would need a security expert in the team, that can evaluate our architecture and analyze where and how we could realize a secure communication.
 I guess the only eCAL transport layer where that makes sense is the tcp layer, and services (which already work over tcp).

If you are looking for secure transportation, you are probably better off looking at other frameworks.
Maybe [this project](https://crossbar.io/autobahn/) could be interesting for you, if you are looking for over the network, secure communications. It also supports Pub/Sub and Client/Server patterns.

**Reply by wbodley** [reply]
Thanks for the info,

We may take a look in the future to see what the possibility is of patching the TCP layer in eCAL with openSSL, will let you know if we do.

Thanks,

Will

**Reply by KerstinKeller** [reply]
For our TCP communication, we use the [tcp_pubsub](https://github.com/eclipse-ecal/tcp_pubsub) project, which is based on asio.
It might be possible to patch that with openSSL without too much hassle.
Please let us know when you do!
